calico/common.py

```diff
@@ -282,6 +282,12 @@ def complete_logging(logfile=None,
             file_handler.setFormatter(formatter)
             root_logger.addHandler(file_handler)
 
+    # Optimization: disable all logging below the minimum level that we care
+    # about.  The global "disable" setting is the first thing that gets checked
+    # in the logging framework so it's the fastest way to disable logging.
+    min_log_level = min(file_level, syslog_level, stream_level)
+    logging.disable(min_log_level - 1)
+
```

Comment:
What happens if one of the levels is "None"?

It turns out that it cannot be (the calling code never sets any level to None). Worth removing the test against file_level being None a few lines above then. Arguably we could assert, but I don't think it is worth it.